utils/http.go

```diff
@@ -325,6 +349,17 @@ func (h *HTTP) EnsureCert(defaultPath string) (err error) {
 
 	log.Warningf("Generating self signed key and cert to %v %v.", h.KeyFile, h.CertFile)
 
+	certDir := path.Dir(defaultCertPath)
+	if _, err = os.Stat(certDir); os.IsNotExist(err) {
```

Comment:
It's rather defensive approach and note that non-existent directory is not the only reason the file write could fail. The file permissions are the reason too, directory could exist but be invisible for current user, entire filesystem could be mount readonly, etc.

I think it's better to just add more detailed error text describing the possible reasons of fail, e.g. change `"error writing key PEM"` to something like `"error writing key PEM, please check that directory exists and has write permissions"`.

Reply:
I think you're right that the whole approach to create the certs path if it's not there is defensive. You're also right that using self-signed certs for plugins even in testing makes less sense than using self-signed certs in Teleport itself, since all of 3rd party service won't work with the self-signed certs anyway. 

In #105, I've just added a task that creates the directories explicitly one chooses to. 

You're also right that it would be an improvement to add some explicit error message in case the directory is not writable or does not exist. 

Creating default directories automatically on `--insecure-no-tls` just makes sure that _if_ the user uses that flag — than the certificates will be generated in 99% situations, unless the dirs exist but the permissions are broken. 

Again, I don't have a strong preference, maybe it's easier to not create the directories, but give an explicit error message and make sure the user understands that they should create a directory, and that self-signed certs will be generated and put there, instead of just silently creating them.

Reply:
> Creating default directories automatically on --insecure-no-tls just makes sure that if the user uses that flag — than the certificates will be generated in 99% situations, unless the dirs exist but the permissions are broken.

`--insecure-no-tls` has nothing to do with self-signed certificates cause it disables TLS at all. This option exposes a plain HTTP port which is useful for development or if you deploy a plugin behind some sort of reverse proxy like Traefik.

Reply:
Lets do both — keep this Mkdir *and* clarify the "pem write error" message.